gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthServer.js

```diff
@@ -506,4 +515,21 @@ gpii.oauth2.authServer.contributeRouteHandlers = function (that, oauth2orizeServ
         }
     );
 
+    that.expressApp.post('/add-preferences',
+        function(req, res) {
+            var accessToken = gpii.oauth2.parseBearerAuthorizationHeader(req);
+            if (!accessToken) {
+                res.send(401);
+            } else {
+                var client = that.authorizationService.getCredentialClientByAccessToken(accessToken);
+                if (!client) {
+                    res.send(404);
+                } else if (!client.allowAddPrefs) {
+                    res.send(401);
+                } else {
+                    that.authorizationService.savePrefs(req, function (data) { res.json(data) });
```

Comment:
Please extract the data from req rather than passing on directly (as discussed below in the comment on authorizationService.savePrefs)